### ISSUE TITLE
gitter for R 4.0.2?

### ISSUE THREAD

**Issue by k-roy** [omarwagih/gitter#5]

**Comment by omarwagih** [comment]
Does it not work as expected now?

**Comment by k-roy** [comment]
Hi Dr. Wagih,

*On R Studio I get this:*

> install.packages('gitter')
Warning in install.packages :
  package ‘gitter’ is not available (for R version 4.0.2)


*From the CRAN link this shows up:*

Package ‘gitter’ was removed from the CRAN repository.

Formerly available versions can be obtained from the archive
<https://cran.r-project.org/src/contrib/Archive/gitter>.

Archived on 2020-01-16 as maintainer's email address bounced and it
requires archived package 'PET'.

Please use the canonical form https://CRAN.R-project.org/package=gitter
<https://cran.r-project.org/package=gitter> to link to this page.

Kevin

**************************************************
*Kevin R. Roy, Ph.D.*
*Postdoctoral Scholar*

*Laboratory of Prof. Lars M. Steinmetz*
*Department of Genetics*
*Stanford University School of Medicine*

*Stanford Genome Technology Center*

*3165 Porter Drive, Room 1220*
*Palo Alto, CA 94304*
*Lab: (650) 721-5542*
*Cell: (310) 991-8127*
*kevinroy@stanford.edu* <kevinroy@stanford.edu>
**************************************************

**Comment by herbstk** [comment]
Since `gitter` as well as its dependency `PET` are now removed from CRAN I installed the packages using:
```
devtools::install_github("cran/PET")
devtools::install_github("omarwagih/gitter")
```
Unfortunately, `PET` can't be compiled with gcc>9 because a multiple variable definition error in this package will not be ignored anymore with newer C compiler versions. I was able to install the package after using a `~/.R/Makevars` file with content `CC=gcc-9` or adding `-fcommon` to variable `CFLAGS`.

**Comment by phancanhtrinh** [comment]
Hi,
I have a similar problem when I install PET. Do you know how can we solve this? 
Thanks so much 
`** libs
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c Hough.c -o Hough.o
Hough.c: In function 'Hough4':
Hough.c:41:10: warning: variable 'theta_min' set but not used [-Wunused-but-set-variable]
   41 |   double theta_min;
      |          ^~~~~~~~~
Hough.c:40:38: warning: variable 'y' set but not used [-Wunused-but-set-variable]
   40 |   double *costheta,*sintheta,theta,x,y;
      |                                      ^
Hough.c: In function 'Hough3':
Hough.c:118:10: warning: variable 'theta_min' set but not used [-Wunused-but-set-variable]
  118 |   double theta_min;
      |          ^~~~~~~~~
Hough.c: In function 'Hough2':
Hough.c:173:10: warning: variable 'theta_min' set but not used [-Wunused-but-set-variable]
  173 |   double theta_min;
      |          ^~~~~~~~~
Hough.c: In function 'Hough1':
Hough.c:227:10: warning: variable 'theta_min' set but not used [-Wunused-but-set-variable]
  227 |   double theta_min;
      |          ^~~~~~~~~
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c PET_init.c -o PET_init.o
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c amatrix.c -o amatrix.o
amatrix.c: In function 'RegulateMatrix':
amatrix.c:234:10: warning: variable 'tempVv' set but not used [-Wunused-but-set-variable]
  234 |   float *tempVv;
      |          ^~~~~~
amatrix.c: In function 'RegulateL1Matrix':
amatrix.c:289:10: warning: variable 'tempVv' set but not used [-Wunused-but-set-variable]
  289 |   float *tempVv;
      |          ^~~~~~
amatrix.c: In function 'GenerateAMatrixColumn':
amatrix.c:784:9: warning: variable 'onedivsq2' set but not used [-Wunused-but-set-variable]
  784 |   float onedivsq2,psi,pidivDeltaX,costheta,sintheta,of;
      |         ^~~~~~~~~
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c art.c -o art.o
art.c: In function 'FAST_ART':
art.c:128:10: warning: variable 'tempXv' set but not used [-Wunused-but-set-variable]
  128 |   float *tempXv,*tempBv;
      |          ^~~~~~
art.c: In function 'SLOW_ART':
art.c:239:10: warning: variable 'tempXv' set but not used [-Wunused-but-set-variable]
  239 |   float *tempXv,*tempBv;
      |          ^~~~~~
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c calc.c -o calc.o
calc.c: In function 'BackFilter':
calc.c:335:19: warning: variable 'M' set but not used [-Wunused-but-set-variable]
  335 |   int i,m,n,mm,nn,M,N,OldHeight,OldWidth;
      |                   ^
calc.c: In function 'CentralSliceCZ':
calc.c:700:49: warning: variable 'rhomin' set but not used [-Wunused-but-set-variable]
  700 |   float DC,deltatheta,theta,thetaadj,*realtheta,rhomin,deltarho;
      |                                                 ^~~~~~
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c cg.c -o cg.o
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c em.c -o em.o
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c eval.c -o eval.o
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c imgtools.c -o imgtools.o
imgtools.c: In function 'ReadPET':
imgtools.c:1005:10: warning: variable 'result' set but not used [-Wunused-but-set-variable]
 1005 |   size_t result;
      |          ^~~~~~
imgtools.c: In function 'ReadFIFHeader':
imgtools.c:1148:10: warning: variable 'result' set but not used [-Wunused-but-set-variable]
 1148 |   size_t result;
      |          ^~~~~~
imgtools.c: In function 'ReadFIF':
imgtools.c:1222:10: warning: variable 'result' set but not used [-Wunused-but-set-variable]
 1222 |   size_t result;
      |          ^~~~~~
imgtools.c: In function 'ReadAnalyze':
imgtools.c:1616:10: warning: variable 'result' set but not used [-Wunused-but-set-variable]
 1616 |   size_t result;
      |          ^~~~~~
imgtools.c:1614:21: warning: variable 'OffSet' set but not used [-Wunused-but-set-variable]
 1614 |   float ScaleFactor,OffSet;
      |                     ^~~~~~
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c iradon.c -o iradon.o
iradon.c: In function 'iradon':
iradon.c:271:5: warning: 'spectrum' may be used uninitialized in this function [-Wmaybe-uninitialized]
  271 |     FreeImage(spectrum);
      |     ^~~~~~~~~~~~~~~~~~~
iradon.c: In function 'loadFile':
iradon.c:568:9: warning: 'list' may be used uninitialized in this function [-Wmaybe-uninitialized]
  568 |   return(list);
      |         ^
iradon.c:404:16: warning: 'typeF' may be used uninitialized in this function [-Wmaybe-uninitialized]
  404 |     OutImage = ReadImage(fileName, typeF);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
iradon.c: In function 'writeFile':
iradon.c:684:19: warning: 'N' may be used uninitialized in this function [-Wmaybe-uninitialized]
  684 |         SaveImage=NewFloatImage(fileName, M, N, _RealArray);
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
iradon.c:684:19: warning: 'M' may be used uninitialized in this function [-Wmaybe-uninitialized]
iradon.c:772:5: warning: 'typeF' may be used uninitialized in this function [-Wmaybe-uninitialized]
  772 |     WriteImage(SaveImage, typeF);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c it.c -o it.o
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c itreadargs.c -o itreadargs.o
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c misc.c -o misc.o
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c oradLI.c -o oradLI.o
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c oradNN.c -o oradNN.o
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c oradSINC.c -o oradSINC.o
oradSINC.c: In function 'radonSINC':
oradSINC.c:52:10: warning: variable 'idx' set but not used [-Wunused-but-set-variable]
   52 |   double idx,mintrig,psi;
      |          ^~~
gcc  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c sparse.c -o sparse.o
sparse.c: In function 'ReadSIFHeader':
sparse.c:607:10: warning: variable 'result' set but not used [-Wunused-but-set-variable]
  607 |   size_t result;
      |          ^~~~~~
sparse.c: In function 'ReadSIF':
sparse.c:657:10: warning: variable 'result' set but not used [-Wunused-but-set-variable]
  657 |   size_t result;
      |          ^~~~~~
sparse.c: In function 'MeanFilterVector':
sparse.c:1537:13: warning: variable 'area' set but not used [-Wunused-but-set-variable]
 1537 |   float sum,area,*tempVv,*tempNv,*tempKernM;
      |             ^~~~
gcc -shared -s -static-libgcc -o PET.dll tmp.def Hough.o PET_init.o amatrix.o art.o calc.o cg.o em.o eval.o imgtools.o iradon.o it.o itreadargs.o misc.o oradLI.o oradNN.o oradSINC.o sparse.o -LC:/rtools42/x86_64-w64-mingw32.static.posix/lib/x64 -LC:/rtools42/x86_64-w64-mingw32.static.posix/lib -LC:/PROGRA~1/R/R-42~1.0/bin/x64 -lR
C:\rtools42\x86_64-w64-mingw32.static.posix\bin/ld.exe: it.o:it.c:(.bss+0x610): multiple definition of `DebugNiveau'; iradon.o:iradon.c:(.data+0x0): first defined here
C:\rtools42\x86_64-w64-mingw32.static.posix\bin/ld.exe: it.o:it.c:(.bss+0x480): multiple definition of `IniFile'; iradon.o:iradon.c:(.bss+0x0): first defined here
C:\rtools42\x86_64-w64-mingw32.static.posix\bin/ld.exe: it.o:it.c:(.bss+0x598): multiple definition of `multtemp'; iradon.o:iradon.c:(.bss+0x118): first defined here
C:\rtools42\x86_64-w64-mingw32.static.posix\bin/ld.exe: it.o:it.c:(.bss+0x5a0): multiple definition of `LogFileName'; iradon.o:iradon.c:(.bss+0x120): first defined here
C:\rtools42\x86_64-w64-mingw32.static.posix\bin/ld.exe: it.o:it.c:(.bss+0x608): multiple definition of `LogFile'; iradon.o:iradon.c:(.bss+0x188): first defined here
collect2.exe: error: ld returned 1 exit status
no DLL was created
ERROR: compilation failed for package 'PET'
* removing 'C:/Users/Trinh/AppData/Local/R/win-library/4.2/PET'
Warning message:
In i.p(...) :
  installation of package ‘C:/Users/Trinh/AppData/Local/Temp/RtmpWSM8Xg/file6b20734d3180/PET_0.5.1.tar.gz’ had non-zero exit status
`

**Comment by omarwagih** [comment]
PET has been causing a lot of issues and is no longer being actively maintained in CRAN. gitter depends on PET for radon-transformation functions (which allows you to autorotate plate images to be perfectly horizontal). I don't think many use this feature so I have removed it in the latest version of gitter (v.1.1.4) and removed the dependency on PET. Setting autorotate=TRUE will throw an error explaining the above.

This should mean that if you try install via github (using `devtools::install_github`), it should work smoothly now. I'll see if I can get it updated on CRAN as well.

From now on, If you want to use the autorotate feature, you'll have to figure out how to manually install PET and install an older version of gitter<=v1.1.3.

Closing this for now as it should address this issue.
